### PR TITLE
[SPARK-5994] [SQL] Python DataFrame documentation fixes

### PR DESCRIPTION
**Pull request by davies**
select empty should NOT be the same as select. make sure selectExpr is behaving the same.
join param documentation
link to source doesn't work in jekyll generated file
cross reference of columns (i.e. enabling linking)
show(): move df example before df.show()
move tests in SQLContext out of docstring otherwise doc is too long
Column.desc and .asc doesn't have any documentation
in documentation, sort functions.*)

### CODE REVIEW

**Comment by davies** [comment]
cc @rxin

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def _(col):
     globals()[_name] = _create_function(_name, _doc)
 del _name, _doc
 __all__ += _functions.keys()
```
Comment (rxin) [inline python/pyspark/sql/functions.py]: can we add a line of comment to __all__ to say that _functions are automatically added to the list.

**Comment by rxin** [comment]
lgtm

**Comment by marmbrus** [comment]
Thanks!  Merged to master and branch 1.3

**Pull request closed without merging**